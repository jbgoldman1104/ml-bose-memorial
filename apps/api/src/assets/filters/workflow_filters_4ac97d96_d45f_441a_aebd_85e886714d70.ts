export const filterSectionFilters = [
  { name: 'TransactionId', statisticalType: 'Data - Categorical - free form' },
  { name: 'InvoiceNumber', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'TotalDonorPayments', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'InvoiceAmount', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'MemberPaidAmount', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'TransactionDonationAmount', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'TransactionPcp', statisticalType: 'Data - Categorical - free form' },
  { name: 'TransactionAmount', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'TransactionDate', statisticalType: 'Data - Numeric - Temporal' },
  { name: 'InvoiceId', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'TransactionDonor', statisticalType: 'Data - Categorical - free form' },
  { name: 'TransactionMember', statisticalType: 'Data - Categorical - free form' },
  { name: 'MemberAmountPaid', statisticalType: 'Data - Numeric - NTG - Discrete' },
];
