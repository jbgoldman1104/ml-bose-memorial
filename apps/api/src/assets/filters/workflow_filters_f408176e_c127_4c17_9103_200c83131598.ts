export const filterSectionFilters = [
  { name: 'DonorId', statisticalType: 'Data - Categorical - free form' },
  { name: 'DonorCommitment', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'DonorFax', statisticalType: 'Data - Categorical - unordered' },
  { name: 'DonorSpentYtd', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'DonorZip', statisticalType: 'Data - Categorical - unordered - geo' },
  { name: 'DonorCurrentBalance', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'MinPerTransaction', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'DonorSuite', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'DonorStreetAddress', statisticalType: 'Data - Categorical - free form' },
  { name: 'PaidMembers', statisticalType: 'Data - Categorical - free form' },
  { name: 'DonorName', statisticalType: 'Data - Categorical - free form' },
  { name: 'MaxPerTransaction', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'DonorPhoneNumber', statisticalType: 'Data - Categorical - unordered' },
  { name: 'BusinessName', statisticalType: 'Data - Categorical - free form' },
  { name: 'PreferredMember', statisticalType: 'Data - Categorical - free form' },
];
