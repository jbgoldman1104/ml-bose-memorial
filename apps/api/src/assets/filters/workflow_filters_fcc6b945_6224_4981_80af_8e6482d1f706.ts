export const filterSectionFilters = [
  { name: 'PrimaryCareId', statisticalType: 'Data - Categorical - free form' },
  { name: 'HcapScore', statisticalType: 'Data - Numeric - NTG - Severity Ranking' },
  { name: 'PcpMember', statisticalType: 'Data - Categorical - free form' },
  { name: 'P2P', statisticalType: 'Data - Categorical' },
  { name: 'CommercialInsurance', statisticalType: 'Data - Categorical' },
  { name: 'BasicFee', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'SpecialistFee', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'OfficeHours', statisticalType: 'Data - Categorical' },
  { name: 'Npi', statisticalType: 'Data - Numeric - NTG - Discrete' },
  { name: 'PcpSuite', statisticalType: 'Data - Categorical' },
  { name: 'PcpStreetAddress', statisticalType: 'Data - Categorical - free form' },
  { name: 'PcpzipCode', statisticalType: 'Data - Categorical - unordered - geo' },
  { name: 'Telephone', statisticalType: 'Data - Categorical - unordered' },
  { name: 'Title', statisticalType: 'Data - Categorical' },
  { name: 'PrimaryCareName', statisticalType: 'Data - Categorical - free form' },
];
