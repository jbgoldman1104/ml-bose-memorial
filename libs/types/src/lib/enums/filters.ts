export enum FilterCombinator {
  AND = 'AND',
  OR = 'OR',
}

export enum FilterOperation {
  GreaterOrEqualThan = '>=',
  Is = '=',
  LessOrEqualThan = '<=',
  RelativeDate = 'relativeDate',
}

/**
 * @deprecated This will be deprecated in coming Releases, please only use it for testing purposes, use FilterOperation instead.
 */
export enum FilterOperators {
  from = 'from',
  in = '_in',
  max = 'max',
  min = 'min',
  relativeDate = 'relativeDate',
  to = 'to',
}
