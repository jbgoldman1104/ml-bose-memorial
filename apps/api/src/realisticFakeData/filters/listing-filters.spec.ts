import {
  FakeDataEntity,
  FilterCombinator,
  FilterOperation,
  FilterQuery,
  IntervalDate,
  Maybe,
  Thing,
} from '@kleeen/types';
import { FakeDataSource, filterList } from './filters';

import mockData from './mocks/listing.mock.json';
import mockThings from './mocks/things.mock.json';

const {
  capuchinoFelizCoffeeShop,
  chocolatito,
  latteMiCorazon,
  macchiato,
  mochachinoDulce,
  seriouslyBlack,
  sinAzucar,
  tripleCafeConLeche,
} = mockData;

export const listingMockData: FakeDataEntity[] = Object.values(mockData);

jest.mock('@kleeen/things', () => {
  return {
    getThingById(id: string): Maybe<Thing> {
      return mockThings[id];
    },
    getThingByName(name: string): Maybe<Thing> {
      return Object.values(mockThings).find((thing) => thing.name === name) as Thing;
    },
  };
});

describe('realisticFakeData', () => {
  const entityName = 'coffeeShop';

  describe('listing filters', () => {
    describe('using unexpected values', () => {
      it('should return the complete list of entities when there are no filter rules', () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData).toEqual(listingMockData);
      });

      it('should return the complete list of entities when the field of the rule is empty', () => {
        const filters = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              operation: FilterOperation.Is,
              value: { displayValue: 'Capuchino Feliz' },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters: filters as FilterQuery,
          list: listingMockData,
        });

        expect(filteredData).toEqual(listingMockData);
      });

      it('should return the complete list of entities when the value of the filter is undefined', () => {
        const filters = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters: filters as FilterQuery,
          list: listingMockData,
        });

        expect(filteredData).toEqual(listingMockData);
      });

      it('should return the complete list of entities when the value of the filter is an empty object', () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: {},
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData).toEqual(listingMockData);
      });

      it('should return the complete list of entities when the operation is missing', () => {
        const filters = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeShop',
              value: { displayValue: 'Capuchino Feliz' },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters: filters as FilterQuery,
          list: listingMockData,
        });

        expect(filteredData).toEqual(listingMockData);
      });

      it('should return the complete list of entities when the operation is not a unknown one', () => {
        const filters = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeShop',
              value: { displayValue: 'Capuchino Feliz' },
              operation: 'absurdFilterOperation',
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters: filters as FilterQuery,
          list: listingMockData,
        });

        expect(filteredData).toEqual(listingMockData);
      });
    });

    describe(`using the ${FilterOperation.Is} operator`, () => {
      it('should filter on an attribute with single values using a single string displayValue', () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { displayValue: 'Capuchino Feliz' },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(1);
        expect(filteredData).toContainEqual(capuchinoFelizCoffeeShop);
      });

      it('should filter on an attribute with single values using a single number displayValue', () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'marketValue',
              operation: FilterOperation.Is,
              value: { displayValue: 8 },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(1);
        expect(filteredData).toContainEqual(capuchinoFelizCoffeeShop);
      });

      it('should filter on an attribute with single values using a single id', () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { id: 'e1c266f8-b31d-47b9-8b22-4cce57ecae1d' },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(1);
        expect(filteredData).toContainEqual(macchiato);
      });

      it('should filter on an attribute with single values using multiple string displayValues', () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: [{ displayValue: 'Capuchino Feliz' }, { displayValue: 'Mocachino Dulce' }],
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(2);
        expect(filteredData).toContainEqual(capuchinoFelizCoffeeShop);
        expect(filteredData).toContainEqual(mochachinoDulce);
      });

      it('should filter on an attribute with single values using multiple number displayValues', () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'marketValue',
              operation: FilterOperation.Is,
              value: [{ displayValue: 8 }, { displayValue: 21 }],
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(2);
        expect(filteredData).toContainEqual(capuchinoFelizCoffeeShop);
        expect(filteredData).toContainEqual(chocolatito);
      });

      it('should filter on an attribute with single values using multiple ids', () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: [
                { id: '26a85695-c945-4d35-bd0b-1a61b236ea09' },
                { id: '545fffcc-59b9-48df-acf4-4db5a1371823' },
                { id: '55e3d48f-4fcb-4212-8fbe-9d678f68b68e' },
              ],
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(3);
        expect(filteredData).toContainEqual(capuchinoFelizCoffeeShop);
        expect(filteredData).toContainEqual(mochachinoDulce);
        expect(filteredData).toContainEqual(seriouslyBlack);
      });
    });

    describe(`using the ${FilterOperation.GreaterOrEqualThan} operator`, () => {
      it('should not filter on an attribute with single values using an unexpected displayValue', () => {
        const filters = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { displayValue: '' },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { displayValue: null },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { displayValue: undefined },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { displayValue: [] },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { displayValue: {} },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters: filters as FilterQuery,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(9);
        expect(filteredData).toEqual(listingMockData);
      });

      it('should filter on an attribute with single values using a single displayValue', () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'marketValue',
              operation: FilterOperation.GreaterOrEqualThan,
              value: { displayValue: 88 },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(2);
        expect(filteredData).toContainEqual(mochachinoDulce);
        expect(filteredData).toContainEqual(tripleCafeConLeche);
      });
    });

    describe(`using the ${FilterOperation.LessOrEqualThan} operator`, () => {
      it('should not filter on an attribute with single values using an unexpected displayValue', () => {
        const filters = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { displayValue: '' },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { displayValue: null },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { displayValue: undefined },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { displayValue: [] },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.Is,
              value: { displayValue: {} },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters: filters as FilterQuery,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(9);
        expect(filteredData).toEqual(listingMockData);
      });

      it('should filter on an attribute with single values using a single displayValue', () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'marketValue',
              operation: FilterOperation.LessOrEqualThan,
              value: { displayValue: 30 },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(5);
        expect(filteredData).toContainEqual(capuchinoFelizCoffeeShop);
        expect(filteredData).toContainEqual(chocolatito);
        expect(filteredData).toContainEqual(macchiato);
        expect(filteredData).toContainEqual(seriouslyBlack);
        expect(filteredData).toContainEqual(sinAzucar);
      });
    });

    describe(`using the ${FilterOperation.RelativeDate} operator`, () => {
      it('should not filter on an attribute with single values using an unexpected relative value', () => {
        const filters = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeShop',
              operation: FilterOperation.RelativeDate,
              value: { displayValue: '' },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.RelativeDate,
              value: { displayValue: null },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.RelativeDate,
              value: { displayValue: undefined },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.RelativeDate,
              value: { displayValue: [] },
            },
            {
              field: 'coffeeShop',
              operation: FilterOperation.RelativeDate,
              value: { displayValue: {} },
            },
            {
              field: 'timestamp',
              operation: FilterOperation.RelativeDate,
              value: { displayValue: 678 },
            },
            {
              field: 'timestamp',
              operation: FilterOperation.RelativeDate,
              value: { displayValue: 'unknownRelativeDate' },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters: filters as FilterQuery,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(9);
        expect(filteredData).toEqual(listingMockData);
      });

      it(`should filter on an attribute with single values using ${IntervalDate.allTime} aka "All Time"`, () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'timestamp',
              operation: FilterOperation.RelativeDate,
              value: { displayValue: IntervalDate.allTime },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(9);
        expect(filteredData).toEqual(listingMockData);
      });

      it(`should filter on an attribute with single values using ${IntervalDate.oneWeek} aka "One Month"`, () => {
        const mockDate = 1633387281359;

        jest.spyOn(global.Date, 'now').mockImplementation(() => mockDate);

        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'timestamp',
              operation: FilterOperation.RelativeDate,
              value: { displayValue: IntervalDate.oneWeek },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(1);
        expect(filteredData).toContainEqual(capuchinoFelizCoffeeShop);
      });

      it(`should filter on an attribute with single values using ${IntervalDate.threeMonth} aka "One Quarter"`, () => {
        const mockDate = 1633387281359;

        jest.spyOn(global.Date, 'now').mockImplementation(() => mockDate);

        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'timestamp',
              operation: FilterOperation.RelativeDate,
              value: { displayValue: IntervalDate.threeMonth },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(2);
        expect(filteredData).toContainEqual(capuchinoFelizCoffeeShop);
        expect(filteredData).toContainEqual(chocolatito);
      });
    });

    describe('using operation combination', () => {
      it('should filter on an attribute with single values using a range operator as single displayValue', () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'managerReview',
              operation: FilterOperation.GreaterOrEqualThan,
              value: { displayValue: 6 },
            },
            {
              field: 'managerReview',
              operation: FilterOperation.LessOrEqualThan,
              value: { displayValue: 12 },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(2);
        expect(filteredData).toContainEqual(capuchinoFelizCoffeeShop);
        expect(filteredData).toContainEqual(latteMiCorazon);
      });

      it(`should filter on an attribute with single values using a date range operator as single displayValue`, () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'timestamp',
              operation: FilterOperation.GreaterOrEqualThan,
              value: { displayValue: 1623247503734 },
            },
            {
              field: 'timestamp',
              operation: FilterOperation.LessOrEqualThan,
              value: { displayValue: 1632688882016 },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(5);
        expect(filteredData).toContainEqual(chocolatito);
        expect(filteredData).toContainEqual(macchiato);
        expect(filteredData).toContainEqual(mochachinoDulce);
        expect(filteredData).toContainEqual(seriouslyBlack);
        expect(filteredData).toContainEqual(sinAzucar);
      });

      it(`should not filter on an attribute with single values using an ${FilterCombinator.AND} combinator with not existing values`, () => {
        const filters: FilterQuery = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'yearlyIncome',
              operation: FilterOperation.Is,
              value: { displayValue: 76 },
            },
            {
              field: 'barista',
              operation: FilterOperation.Is,
              value: { id: '898b2087-dda0-478a-91fe-a00ceb29b8a2' },
            },
          ],
        };

        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData.length).toBe(9);
        expect(filteredData).toEqual(listingMockData);
      });

      it.todo(`should filter on an attribute with single values using an ${FilterCombinator.OR} combinator`);

      it('should filter on an attribute with single values using a combination of operations', () => {
        const filters = {
          combinator: FilterCombinator.AND,
          rules: [
            {
              field: 'coffeeQuality',
              operation: FilterOperation.Is,
              value: [
                {
                  displayValue: 'Gross',
                },
              ],
            },
            {
              field: 'marketValue',
              operation: FilterOperation.LessOrEqualThan,
              value: {
                displayValue: 10,
              },
            },
            {
              field: 'marketValue',
              operation: FilterOperation.GreaterOrEqualThan,
              value: {
                displayValue: 0,
              },
            },
            {
              field: 'timestamp',
              operation: FilterOperation.GreaterOrEqualThan,
              value: {
                displayValue: 1623247503734,
              },
            },
            {
              field: 'timestamp',
              operation: FilterOperation.LessOrEqualThan,
              value: {
                displayValue: 1623247503734,
              },
            },
          ],
        };
        const filteredData = filterList({
          entityName,
          fakeDataSource: FakeDataSource.ListingData,
          filters,
          list: listingMockData,
        });

        expect(filteredData).toMatchSnapshot();
      });
    });
  });
});
