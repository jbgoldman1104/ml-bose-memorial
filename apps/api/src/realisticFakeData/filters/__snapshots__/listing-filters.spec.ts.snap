// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`realisticFakeData listing filters using operation combination should filter on an attribute with single values using a combination of operations 1`] = `
Array [
  Object {
    "barista": Object {
      "displayMedia": "",
      "id": "682e239d-e3f5-4da8-a524-d679dd1f5ad3",
    },
    "coffee": Object {
      "displayMedia": "",
      "id": "1858053e-a2bf-47c7-8403-0e68df57065f",
    },
    "coffeeQuality": Object {
      "displayMedia": Object {
        "type": "svg",
        "value": "https://raw.githubusercontent.com/KLEEEN-SOFTWARE/Kleeen-svgs/main/assets/severity-levels-svgs/severity-level-1of4.svg",
      },
      "displayValue": "Gross",
      "id": "fa53894b-3420-471d-a961-66d9f647033d",
    },
    "coffeeShop": Object {
      "displayMedia": Object {
        "type": "text",
        "value": "Machiatto",
      },
      "displayValue": "Machiatto",
      "id": "e1c266f8-b31d-47b9-8b22-4cce57ecae1d",
    },
    "customer": Object {
      "displayMedia": "",
      "id": "ebaab08d-6b34-47b6-ac2e-d3f78aca8130",
    },
    "description": Object {
      "displayMedia": "",
      "displayValue": "Laboriosam eligendi aliquam dicta rerum. Quia voluptatem est dolores voluptatem provident iusto id.",
      "id": "e39f8999-5f8e-4fd7-bdd1-7b9ec7b13d70",
    },
    "displayValue::coffeeShop": Object {
      "displayValue": "Machiatto",
      "id": "e1c266f8-b31d-47b9-8b22-4cce57ecae1d",
    },
    "efficiencyPercent": Object {
      "displayMedia": "",
      "id": "792aeedf-63ac-4a3d-8537-9d5627fe21f2",
    },
    "id": "e1c266f8-b31d-47b9-8b22-4cce57ecae1d",
    "importantEvent": Object {
      "displayMedia": "",
      "id": "d8861460-1ac5-4cd6-b8ac-96aedb003fae",
    },
    "inauguration": Object {
      "displayMedia": "",
      "displayValue": 1629062535111,
      "id": "bb75364c-115f-4e40-b477-782241eb540b",
    },
    "isOpen": Object {
      "displayMedia": "",
      "displayValue": false,
      "id": "edf129ec-c3eb-4887-9daf-8837ead1d027",
    },
    "managerReview": Object {
      "displayMedia": "",
      "displayValue": 19,
      "id": "2ed6e6e1-090c-4080-b038-6a260be4f3ca",
    },
    "marketValue": Object {
      "displayMedia": "",
      "displayValue": 5,
      "id": "09abe778-7045-41c8-9d1b-c1f80e5748d1",
    },
    "menu": Object {
      "displayMedia": "",
      "displayValue": 940,
      "id": "835a9201-a5e4-42d3-a933-2d640c85cdff",
    },
    "monthlyBrandingColor": Object {
      "displayMedia": "",
      "displayValue": "rgba(190,134,190,0.72)",
      "id": "ca45667b-c1b0-4611-bb5e-dfb124a8a261",
    },
    "occupationPercent": Object {
      "displayMedia": "",
      "displayValue": 31,
      "id": "6b1d1c3d-51df-4a3a-969a-06f03a9556c8",
    },
    "previousBrandingColors": Array [
      Object {
        "displayValue": "rgba(151,248,39,0.33)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(128,103,200,0.11)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(188,39,217,0.9)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(100,239,157,0.71)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(36,223,195,0.96)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(199,253,4,0.06)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(86,56,24,0.4)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(112,109,140,0.25)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(117,54,71,0.66)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(95,104,198,0.77)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(16,124,185,0.73)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(76,32,225,0.53)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(33,193,153,0.03)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(102,130,190,0.29)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(135,113,55,0.94)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(242,138,180,0.84)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(3,232,77,0.47)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(239,126,184,0.22)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(75,179,35,0.57)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
      Object {
        "displayValue": "rgba(210,169,26,0.33)",
        "id": "c07bcdb9-bedc-4dbc-9c89-58e3b1254e16",
      },
    ],
    "reviewLevel": Object {
      "displayMedia": "",
      "id": "d8476712-60bb-409e-a06d-b40c27d53d17",
    },
    "reviewScore": Object {
      "displayMedia": "",
      "id": "075aea9a-b7a2-4cea-a72f-87ad6f212bfd",
    },
    "server": Array [
      Object {
        "displayValue": "Angelica",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Gilberto",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Gabe",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Jaqueline",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Zula",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Christian",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Theodore",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Dorcas",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Terrill",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Emerald",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Lenore",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Garfield",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Emma",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Ryleigh",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Tristian",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Marilyne",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Trent",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Hildegard",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Gracie",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
      Object {
        "displayValue": "Kaylee",
        "id": "cde61813-7185-47e3-86dd-7ef7cc5c6728",
      },
    ],
    "squareFootage": Object {
      "displayValue": 1333.65946309641,
      "id": "f202e121-a44b-4bf3-9643-3cf536a3c579",
    },
    "table": Object {
      "displayValue": 40,
    },
    "timestamp": Object {
      "displayValue": 1623247503734,
      "id": "33a31b40-1233-448f-972e-883a54649e32",
    },
    "userReview": Object {
      "displayMedia": "",
      "id": "3ee47dd1-ac1a-4321-ab0c-a458385897ed",
    },
    "userWrittenReviews": Object {
      "displayMedia": "",
      "id": "7aa8a922-dc06-4321-ae0b-16f431fca187",
    },
    "yearlyIncome": Object {
      "displayValue": 76,
    },
  },
]
`;
